src/lib/Herrera/Version/Parser.php

```diff
@@ -73,7 +73,7 @@ public static function toComponents($version)
         }
 
         if (false !== strpos($version, '-')) {
-            list($version, $pre) = explode('-', $version);
+            list($version, $pre) = preg_split('/-/', $version, 1);
```

Comment:
Warning, your limit parameter is incorrect :
```php
<?php
print_r(preg_split('/-/', '1.2.3-pre-release', 1));
?>
```
The above example will output:
```
Array
(
    [0] => 1.2.3-pre-release
)
```
But, fix limit to 2 :
```php
<?php
print_r(preg_split('/-/', '1.2.3-pre-release', 2));
?>
```
The above example will output:
```
Array
(
    [0] => 1.2.3
    [1] => pre-release
)
```
Expected return for the need of your pull request